10_Code/Population_Cleaning.py

```diff
@@ -0,0 +1,111 @@
+"""
+The population data has been taken from https://wonder.cdc.gov/bridged-race-population.html
+The data is in the form of tab-separated values, with additional info towards the end of the document, starting with the text "---"
+We have downloaded the data from 1990 to 2010 separately, and then 2010 to 2020 separately, to maintain the file sizes and ensure complete download of data. We will combine the two in this code.
+We will also remove the additional info at the end of the document while reading the file.
+The raw downloaded data has been saved in the folder "00_raw_data" as "pop_1990_2010.txt" and "pop_2010_2020.txt" 
+"""
+# Importing libraries
+import pandas as pd
+import numpy as np
+
+# Individual file links in the 00_raw_data folder
+data_1990_2010 = "https://github.com/MIDS-at-Duke/opioid-2023-group-8-final-opioid/raw/population_cleaning/00_Source_Data/US_Pop_1990_2010.txt"
+data_2010_2020 = "https://github.com/MIDS-at-Duke/opioid-2023-group-8-final-opioid/raw/population_cleaning/00_Source_Data/US_Pop_2010_2020.txt"
+
+# Reading the file
+# We will read the files in chunks of one line, and stop reading when we encounter the text "---"
+df_1990_2010 = pd.DataFrame()
+df_2010_2020 = pd.DataFrame()
+
+
+def stop_reading(line):
+    return "---" not in line
+
+
+reader = pd.read_csv(data_1990_2010, iterator=True, chunksize=1, sep="\t")
+for chunk in reader:
+    if stop_reading(chunk.to_string()):
+        df_1990_2010 = pd.concat([df_1990_2010, chunk])
+    else:
+        break
+
+reader = pd.read_csv(data_2010_2020, iterator=True, chunksize=1, sep="\t")
+for chunk in reader:
+    if stop_reading(chunk.to_string()):
+        df_2010_2020 = pd.concat([df_2010_2020, chunk])
+    else:
+        break
+
+df_1990_2010.shape
+# (66129, 8)
```

Comment:
Put some important results as comments is also my take-away from your code. It is also helpful for me to understand your code.